plugins/wdc/wdc-nvme.c

```diff
@@ -2114,22 +2115,23 @@ static int wdc_do_cap_dui(int fd, char *file, __u32 xfer_size, int data_area, in
 		}
 	}
 	else if ((log_hdr->hdr_version & 0xFF) == 0x04)	{
-		__s32 log_size = 0;
-		__u32 curr_data_offset = 0;
+		__s64 log_size = 0;
+		__u64 curr_data_offset = 0;
 		struct wdc_dui_log_hdr_v4 *log_hdr_v4;
 		log_hdr_v4 = (struct wdc_dui_log_hdr_v4 *)log_hdr;
+		__u64 xfer_size_long = (__u64)xfer_size;
 
-
-		cap_dui_length = le32_to_cpu(log_hdr_v4->log_size_sectors) * WDC_NVME_SN730_SECTOR_SIZE;
+		cap_dui_length_v4 = le64_to_cpu(log_hdr_v4->log_size_sectors) * WDC_NVME_SN730_SECTOR_SIZE;
 
 		if (verbose) {
 			fprintf(stderr, "INFO : WDC : Capture V4 Device Unit Info log, data area = %d\n", data_area);
 			fprintf(stderr, "INFO : WDC : DUI Header Version = 0x%x\n", log_hdr_v4->hdr_version);
 			fprintf(stderr, "INFO : WDC : DUI Product ID = 0x%x/%c\n", log_hdr_v4->product_id, log_hdr_v4->product_id);
-
+			fprintf(stderr, "INFO : WDC : DUI log size sectors = 0x%x\n", log_hdr_v4->log_size_sectors);
+			fprintf(stderr, "INFO : WDC : DUI cap_dui_length = 0x%lx\n", (unsigned long int)cap_dui_length_v4);
```

Comment:
If it's a 64-bit value, casting to long on a 32-bit arch is usually going to yield a 32-bit value.

Reply:
Thanks for the comment Keith.  I probably should verify these changes on a 32 bit machine (if I can find one).   I don't think we have any in our stock of test systems.